core/platform/datastore/gae_datastore_services.py

```diff
@@ -78,3 +89,78 @@ def fetch_multiple_entities_by_ids_and_models(ids_and_models):
         start_index = start_index + len(entity_ids)
 
     return all_models_grouped_by_model_type
+
+
+def make_pseudo_random_hr_consistency_policy():
+    """Returns a policy that always gives the same sequence of consistency
+    decisions.
+
+    Returns:
+        datastore_stub_util.PseudoRandomHRConsistencyPolicy. The policy.
+    """
+    return datastore_stub_util.PseudoRandomHRConsistencyPolicy(probability=1)
+
+
+@contextlib.contextmanager
+def mock_datetime_for_datastore(mocked_now):
```

Comment:
@brianrodri -- just an FYI, @DubeySandeep was mentioning that it seems weird for this to be in core/platform, because it's a mock that's supposed to be only for tests. Do you think we can somehow find a way to keep the mock in test_utils and only using datastore_services for the ndb-specific parts?

(I think @vojtechjelinek might have a similar concern.)

Reply:
I tried to bypass the mocking by swapping out the implementation of `datetime.datetime.utcnow`, but get an Exception: can't set attributes of built-in/extension type 'datetime.datetime'. This means our only option is to swap out the datetime class itself, but doing so requires us to change some internals of the datastore (seen later on). Given these internals are so hacky and GAE-specific, I think it needs to be factored out as part of the migration process, so that we can find the correct alternative.

Reply:
@DubeySandeep @vojtechjelinek -- any thoughts? I would defer to your opinions.

Reply:
I agree that it would be better to keep it in test_utils, but I think the approach that @brianrodri suggested is fine, so keep it in the `datastore_services` for now and resolve this after the migration.

Reply:
SGTM